### DISCUSSION TITLE
use axios as a graphql client

### DISCUSSION THREAD

**Discussion by joeynimu** [dotansimha/graphql-code-generator#5691] [Q&A]
Is it possible to pass an Axios instance as the graphqlClient to the `getSdk` function? 

That is something like; 

```
const axiosInstance = axios.create({ baseURL: process.env.NEXT_PUBLIC_BASE_API_URL });

// then pass it like so;

const sdk = getSdk(axiosInstance)
```

I am currently using `graphql-request,` but I have this [issue](https://github.com/prisma-labs/graphql-request/issues/255) which is straight forward to handle when using axios. It would be nice if I can use this. If anyone has some info on this, I'd appreciate it. Thanks

**Comment by tvvignesh** [comment]
@joeynimu "Technically" should be possible by adding a sort of adapter on top and adding all the GraphQL clients as plugins to it, but it is currently not supported.

@dotansimha Do you think this is a case which we should support?

**Reply by dotansimha** [reply]
You can use `typescript-generic-sdk` and provide custom executor :)
